test/e2e/cypress/e2e/host_details.cy.js

```diff
@@ -355,4 +355,42 @@ context('Host Details', () => {
       cy.get('span').find('svg').should('exist');
     });
   });
+
+  describe('Deregistration', () => {
+    describe('"Clean up" button should be visible only for an unhealthy host', () => {
+      it('should not display the "Clean up" button for healthy host', () => {
+        cy.task('startAgentHeartbeat', [selectedHost.id]);
+        cy.contains('button', 'Clean up').should('not.exist');
+      });
+
+      it('should show the "Clean up" button once heartbeat is lost and debounce period has elapsed', () => {
+        cy.task('stopAgentsHeartbeat');
+        cy.contains('button', 'Clean up', { timeout: 15000 }).should('exist');
+      });
+    });
+
+    describe('"Clean up" button should deregister a host', () => {
+      before(() => {
+        cy.task('stopAgentsHeartbeat');
+      });
+
+      it('should allow to deregister a host after clean-up confirmation', () => {
+        cy.contains('button', 'Clean up', { timeout: 15000 }).click();
+
+        cy.get('#headlessui-portal-root').as('modal');
```

Comment:
Regarding my previous comment: I think this is how you solved the `contains` getting the `Clean up` button from the modal. If that's so, then I'm up for it. I'll create a tech debt to update the other deregistration e2e tests.